engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/source/TableSourceFactoryMixin.scala

```diff
@@ -0,0 +1,36 @@
+package pl.touk.nussknacker.engine.flink.table.source
+
+import org.apache.flink.api.common.functions.{RichMapFunction, RuntimeContext}
+import org.apache.flink.configuration.Configuration
+import pl.touk.nussknacker.engine.api.Context
+import pl.touk.nussknacker.engine.api.process.{BasicContextInitializer, ContextInitializer, ContextInitializingFunction}
+import pl.touk.nussknacker.engine.api.runtimecontext.EngineRuntimeContext
+import pl.touk.nussknacker.engine.api.typed.typing.Typed
+
+trait TableSourceFactoryMixin {
```

Comment:
I don't like the cake pattern (compositions with mixins). I'd rather prefer standard objects and standard composition. I know that it is no the same as typical inheritance when you have many layers and nobody know what should be where. But take a look at `KafkaUniversalComponentTransformer` or `ConfluentKafkaAvroSerializer` extending `AbstractConfluentKafkaAvroSerializer` which mixin `DatumReaderWriterMixin`. Is it easy to understand for you? :) I'm not forcing the approach because it is rather the matter or sense of taste, I'm very biased towards composition.

Reply:
Yes, complex mixins are hard to understand - I've simplified this

Reply:
Ok, I get it now :D I am kind of familiar with it, I already saw (and probably began) problems with rigid class hierarchies. Don't know why I default to inheritance. Changed it :+1: